src/isomorphic/classic/class/ReactClass.js

```diff
@@ -778,8 +778,11 @@ var ReactClass = {
         // We allow auto-mocks to proceed as if they're returning null.
         if (initialState === undefined &&
             this.getInitialState._isMockFunction) {
-          // This is probably bad practice. Consider warning here and
-          // deprecating this convenience.
+          warning(
+            typeof initialState === 'object' && initialState == null && !Array.isArray(initialState),
```

Comment:
We already know that `initialState === undefined` so this condition will never be met and we'll always warn. It's also an entirely unactionable warning as this only happens when a component is mocked.